client/state/themes/selectors.js

```diff
@@ -602,6 +602,23 @@ export function isPremiumThemeAvailable( state, themeId, siteId ) {
 }
 
 /**
+ * Whether a given theme is installed or can be installed on a Jetpack site.
+ *
+ * @param  {Object}  state   Global state tree
+ * @param  {String}  themeId Theme ID for which we check availability
+ * @param  {Number}  siteId  Site ID
+ * @return {Boolean}         True if siteId is a Jetpack site on which theme is installed or can be installed.
+ */
+export function isThemeAvailableOnJetpackSite( state, themeId, siteId ) {
```

Comment:
You mean just 

```js
config.isEnabled( 'manage/themes/upload' ) && hasJetpackSiteJetpackThemesExtendedFeatures( state, siteId )
```

? Well the thing is, the other checks should be innocuous -- the `isWpcomTheme` just checks if the theme is in the `wpcom` subtree (which it will be, or we couldn't click on it in multi-site mode), and the `getTheme` just checks if it's installed on the JP site already anyway.

Requiring `isWpcomTheme` along with the `...ExtendedFeatures` check is essentially just stating that if we do click on 'try&customize` for a given site, our corresponding action will attempt to install the theme from WP.com if it isn't there already.